system/BaseModel.php

```diff
@@ -995,9 +996,9 @@ public function delete($id = null, bool $purge = false)
 
     /**
      * Permanently deletes all rows that have been marked as deleted
-     * through soft deletes (deleted = 1)
+     * through soft deletes (deleted = 1).
      *
-     * @return mixed
+     * @return bool|string Returns a string if in test mode.
```

Comment:
These are annoying to support. I would like to see in v5 some kind of Database Proxy that passes requests through to the driver for most cases but "captures" the actual SQL query for Test Mode. I think that would be a much better approach than trying to adjust all our return types.

Reply:
Yes, the return types are not good, should be `bool` only.